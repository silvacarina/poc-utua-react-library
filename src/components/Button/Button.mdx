import Button from "./index"

<Button label="Teste" />