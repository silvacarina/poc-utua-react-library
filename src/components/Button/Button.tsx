import React from 'react';
import classNames from 'classnames';
import style from './style.module.css';

// Definindo a interface para as propriedades
interface ButtonProps {
  className?: string;
  type?: 'button' | 'submit' | 'reset';
  label?: string | number;
  variant?: 'primary' | 'secondary' | 'success' | 'danger';
  format?: 'standard' | 'outline' | undefined;
  size?: 's' | 'm' | 'l';
  disabled?: boolean;
  icon?: string;
  iconClassName?: string;
  iconSize?: 'xxxs' | 'xxs' | 'xs' | 's' | 'm' | 'l' | 'xl' | 'xxl' | 'xxxl';
  onClick?: React.MouseEventHandler<HTMLButtonElement>;
  element?: React.ElementType;
  [x: string]: unknown; // para propriedades adicionais
}

export const Button = React.forwardRef<HTMLButtonElement, ButtonProps>(function ButtonComponent(
  {
    className,
    type,
    label,
    variant = 'primary',
    format = 'standard',
    size = 's',
    disabled = false,
    onClick,
    element: ButtonElement = 'button',
    ...rest
  },
  ref,
) {
  const buttonClassNames = classNames(
    style.button,
    style[`type-${variant}`],
    disabled ? style.standard : style[format],
    style[`size-${size}`],
    {
      [style.disabled]: disabled,
      [style.hasLabel]: Boolean(label),
    },
  )

  return (
    <ButtonElement
      type={type}
      onClick={onClick}
      className={classNames(buttonClassNames, className)}
      disabled={disabled}
      ref={ref}
      {...rest}
    >
      {label}
    </ButtonElement>
  )
})
