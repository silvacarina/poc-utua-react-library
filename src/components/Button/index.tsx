import style from './style.module.css'

type ButtonProps = {
  label: string,
  type?: "button" | "submit" | "reset"
}

export default function Button({ label, type = 'button' }: ButtonProps) {
  return (
    <button className={style.button} type={type}>{label}</button>
  )
}