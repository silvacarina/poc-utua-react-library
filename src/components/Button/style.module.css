.button {
  height: max-content;
  display: flex;
  align-items: center;
  justify-content: center;
  border: none;
  border-radius: calc(8px / 2);
  font-weight: 600;
  text-decoration: none;
  cursor: pointer;
  transition: all 200ms ease;
}

.type-primary {
  color: #fdf3e1;
  background: #D03801;
}

.type-secondary {
  color: #344054;
  background: white;
}

.type-success {
  color: #f3fcfa;
  background: #00d68f;
}

.type-danger {
  color: #fef5f7;
  background: #dd1212;
}

.size-white {
  font-size: 14px;
  line-height: 115%;
  padding: 0;
}

.size-s {
  font-size: 14px;
  line-height: 115%;
  padding: 8px 24px;
}

.size-m {
  font-size: 16px;
  line-height: 100%;
  padding: calc(8px + 4px) 32px;
}

.size-l {
  font-size: 16px;
  line-height: 150%;
  padding: calc(8px + 4px) 32px;
}

/* hover, focus and active status */
.type-primary:hover,
.type-primary:focus,
.type-primary:active {
  background: #ec8303;
  box-shadow: 0 4px 8px 0 rgba(247, 158, 27, 0.25);
}

.type-secondary:hover,
.type-secondary:focus,
.type-secondary:active {
  background: #362250;
  box-shadow: 0 4px 8px 0 rgba(42, 19, 85, 0.25);
}

.type-success:hover,
.type-success:focus,
.type-success:active {
  background: #00b887;
  box-shadow: 0 4px 8px 0 rgba(0, 74, 66, 0.25);
}

.type-danger:hover,
.type-danger:focus,
.type-danger:active {
  background: #c21010;
  box-shadow: 0 4px 8px 0 rgba(115, 9, 9, 0.25);
}

/* disable status */
.button:disabled {
  pointer-events: none;
  opacity: 0.5;
}

.type-primary:disabled {
  color: #e87201;
  background: #fdf3e1;
}

.type-secondary:disabled {
  color: #2b1c3f;
  background: #eeecf2;
}

.type-success:disabled {
  color: #004a42;
  background: #f3fcfa;
}

.type-danger:disabled {
  color: #730909;
  background: #fef5f7;
}

/* outline */
.outline, .outline:hover, .outline:focus, .outline:active {
  background: transparent;
}

.type-primary.outline {
  color: #f4ad2b;
  border: 1px solid #f4ad2b;
}

.type-primary.outline:hover,
.type-primary.outline:focus,
.type-primary.outline:active {
  color: #e87201;
  border-color: #e87201;
  box-shadow: 0px 4px 8px 0px rgba(247, 158, 27, 0.35);
}

.type-secondary.outline {
  color: #745999;
  border: 1px solid #745999;
}

.type-secondary.outline:hover,
.type-secondary.outline:focus,
.type-secondary.outline:active {
  color: #2b1c3f;
  border-color: #2b1c3f;
  box-shadow: 0 4px 8px 0 rgba(42, 19, 85, 0.35);
}

.type-success.outline {
  color: #00d68f;
  border: 1px solid #00d68f;
}

.type-success.outline:hover,
.type-success.outline:focus,
.type-success.outline:active {
  color: #004a42;
  border-color: #004a42;
  box-shadow: 0 4px 8px 0 rgba(0, 74, 66, 0.35);
}

.type-danger.outline {
  color: #dd1212;
  border: 1px solid #dd1212;
}

.type-danger.outline:hover,
.type-danger.outline:focus,
.type-danger.outline:active {
  color: #730909;
  border-color: #730909;
  box-shadow: 0 4px 8px 0 rgba(115, 9, 9, 0.35);
}