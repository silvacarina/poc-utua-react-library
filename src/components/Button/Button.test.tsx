import renderer from 'react-test-renderer'
import { describe, expect, it } from 'vitest'
import Button from './index'

describe('Button component', () => {
  it('Button component renders correctly', () => {
    const component = renderer.create(
      <Button label="Submit" />
    )

    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('prop working', () => {
    const component = renderer.create(
      <Button type="reset" label="Reset" />
    )

    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})