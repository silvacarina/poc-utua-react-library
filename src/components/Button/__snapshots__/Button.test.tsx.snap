// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button component > Button component renders correctly 1`] = `
<button
  className="_button_0696ee"
  type="button"
>
  Submit
</button>
`;

exports[`Button component > prop working 1`] = `
<button
  className="_button_0696ee"
  type="reset"
>
  Reset
</button>
`;
